### PR TITLE
Add sassc 3.6.2

### PR DESCRIPTION
**Pull request by lazka**
This is a direct copy of https://github.com/lazka/sassc/tree/meson which I've created for gtk back in the days.

This project is in a similar situation as #1153. There exists an upstream PR for adding meson (https://github.com/sass/sassc/pull/269), but the project is unmaintained, so unlikely to see progress there.

### CODE REVIEW

**Comment by lazka** [comment]
The next step after this would be to try to get rid of all references of my forks in GNOME and replace them with wrapdb: https://github.com/search?q=org%3AGNOME+https%3A%2F%2Fgithub.com%2Flazka%2Fsassc&type=code

File: subprojects/packagefiles/sassc/meson.build [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+sassc_sources = ['sassc.c']
+
+incs = []
+if host_machine.system() == 'windows'
+  windows = import('windows')
+  win_res = windows.compile_resources(
+    'res/libsass.rc',
+    depend_files: ['res/libsass.ico'])
+  sassc_sources += [win_res]
+  incs += [include_directories(join_paths('win', 'posix'))]
```
Comment (jpakkane) [inline subprojects/packagefiles/sassc/meson.build:35]: You don't need to wrap `include_directories` with brackets. You also don't need to use `join_paths` at all. So this should be:

   incs += include_directories('win/posix')

And the same below for `getopt.c`.

**Comment by lazka** [comment]
Thanks!

**Comment by lazka** [comment]
> The next step after this would be to try to get rid of all references of my forks in GNOME and replace them with wrapdb: https://github.com/search?q=org%3AGNOME+https%3A%2F%2Fgithub.com%2Flazka%2Fsassc&type=code

Turns out both projects in GNOME using this depend on the wraps being used in flatpak as separate modules and not within meson, and I couldn't figure out how to use wrapdb without meson in flatpaks, so this project is currently on ice.

**Pull request merged**